openfe/setup/methods/openmm/equil_rbfe_methods.py

```diff
@@ -0,0 +1,533 @@
+       etc...
+
+    Attributes
+    ----------
+    minimization_steps : int
+      Number of minimization steps to perform. Default 10000.
+    equilibration_length : float * unit.picosecond
+      Length of the equilibration phase in units of time.
+    production_length : float * unit.picosecond
+      Length of the production phase in units of time.
+    output_filename : str
+      Path to the storage file for analysis. Default 'rbfe.nc'.
+    checkpoint_interval : int * unit.timestep
+      Frequency to write the checkpoint file. Default 50 * unit.timestep
+    checkpoint_storage : str
+      Optional separate filename for the checkpoint file. Note, this should
+      not be a full path, just a filename. If None, the checkpoint will be
+      written to the same file as output_filename. Default None.
+    """
+    class Config:
+        arbitrary_types_allowed = True
+
+    minimization_steps = 10000
+    equilibration_length: unit.Quantity
+    production_length: unit.Quantity
+
+    # reporter settings
+    output_filename = 'rbfe.nc'
+    checkpoint_interval = 50 * unit.timestep
+    checkpoint_storage = Union[str, None]
+
+    @validator('equilibration_length', 'production_length')
+    def is_time(cls, v):
+        # these are time units, not simulation steps
+        if not v.is_compatible_with(unit.picosecond):
+            raise ValueError("Durations must be in time units")
+        return v
+
+    @validator('minimization_steps', 'equilibration_length',
+               'production_length', 'checkpoint_interval')
+    def must_be_positive(cls, v):
+        if v <= 0:
+            errmsg = ("Minimization steps, MD lengths, and checkpoint "
+                      "intervals must be positive")
+            raise ValueError(errmsg)
+        return v
+
+
+class RelativeLigandTransformSettings(BaseModel):
+    class Config:
+        arbitrary_types_allowed = True
+
+    # Things for creating the systems
+    system_settings: SystemSettings
+    topology_settings: TopologySettings
+
+    # Alchemical settings
+    alchemical_settings: AlchemicalSettings
+
+    # MD Engine things
+    engine_settings = OpenMMEngineSettings()
+
+    # Sampling State defining things
+    integrator_settings: IntegratorSettings
+    barostat_settings: BarostatSettings
+    sampler_settings: SamplerSettings
+
+    # Simulation run settings
+    simulation_settings: SimulationSettings
+
+    # solvent model?
+    solvent_padding = 1.2 * unit.nanometer
+
+
+class RelativeLigandTransformResults:
+    """Dict-like container for the output of a RelativeLigandTransform"""
+    def __init__(self, settings: RelativeLigandTransformSettings):
+        self._parent_settings = settings
+        fn = self._parent_settings.simulation_settings.output_filename
+        self._reporter = multistate.MultiStateReporter(fn)
+        self._analyzer = multistate.MultiStateSamplerAnalyzer(self._reporter)
+
+    def dG(self):
+        """Free energy difference of this transformation
```

Comment:
Here and below, is it worth mentioning we're returning in quantities but default kcal/mol?